### PR TITLE
add new headers

### PR DESCRIPTION
**Pull request by ndurell**
* mobile header indicating traffic coming from mobile sdk
* retry attempt header indicating number of retries

# Description

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

File: Sources/KlaviyoSwift/KlaviyoAPI.swift [hunk 1]
```diff
@@ -49,6 +49,7 @@ struct KlaviyoAPI {
             requestFailed(request, error, 0.0)
             return .failure(.internalRequestError(error))
         }
+        urlRequest.allHTTPHeaderFields?["X-Klaviyo-Retry-Attempt"] = "\(attemptNumber)/50"
```
Comment (ndurell) [inline Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: maybe could use a constant here.

Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: Just to be super sure, this header should be formatted `1/50` -> `50/50` ?

Comment (ndurell) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: yah that was my understanding...could totally have missed that detail though!

Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: Makes sense to me, just want to be sure we don't accidentally start at 0 (or end at 49)

Comment (ndurell) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: Yah I'm going to test it through and ensure it does the desired thing.

Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: I wonder if the backend could be able to reply with a header to tell you to increase your max attempts.

Comment (ndurell) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: @jason-myers-klaviyo let us know if there are specific requirements but IIRC you wanted it to send the last request with 50/50 right?

Comment (jason-myers-klaviyo) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: Yes, I think that's right.  If we see a 50/50 and we return a 429, we'll know that the request will not be tried again.

50/50 meaning "we're on attempt number 50, and we will stop after 50 (meaning we won't go again after this one"

And 1/50 meaning "we're on the first attempt of 50 possible attempts"

**Comment by evan-masseau** [comment]
For viz, the related [android PR](https://github.com/klaviyo/klaviyo-android-sdk/pull/149)

File: Sources/KlaviyoSwift/KlaviyoAPI.swift (hunk 1 above)
Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: Lets name this header `X-Klaviyo-Attempt-Count` then?

Comment (jason-myers-klaviyo) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: I think we landed on

`X-Klaviyo-Retry-Attempt`

Earlier

Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: Yeah its a nitpick, but just responding to the above thread, that name is a little ambiguous: is `1/50` the first _retry_ or first _attempt_.

Comment (jason-myers-klaviyo) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:52]: I see, no issues here then

**Review by evan-masseau — CHANGES_REQUESTED** [review]
Rename to `X-Klaviyo-Attempt-Count`

**Comment by ajaysubra** [comment]
> Rename to `X-Klaviyo-Attempt-Count`

Done. Let me know if it looks good.

File: Sources/KlaviyoSwift/KlaviyoAPI.swift (hunk 1 above)
Comment (ajaysubra) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift]: Starting from 1 now. Still doing some testing on my end but so far it's looking good.

Also, renamed header to `X-Klaviyo-Attempt-Count`

**Review by ajaysubra — APPROVED** [review]

File: Tests/KlaviyoSwiftTests/__Snapshots__/KlaviyoAPITests/testSuccessfulResponseWithEvent.1.txt [hunk 2]
```diff
@@ -9,7 +9,10 @@
   ▿ httpMethod: Optional<String>
     - some: "POST"
   ▿ allHTTPHeaderFields: Optional<Dictionary<String, String>>
-    - some: 0 key/value pairs
+    ▿ some: 1 key/value pair
+      ▿ (2 elements)
+        - key: "X-Klaviyo-Retry-Attempt"
+        - value: "0/50"
```
Comment (evan-masseau) [inline Tests/KlaviyoSwiftTests/__Snapshots__/KlaviyoAPITests/testSuccessfulResponseWithEvent.1.txt:15]: When an actual device sends, the first send will be 1/50, ya?
I had to do this in my tests too, just want to confirm.

Comment (ajaysubra) [reply Tests/KlaviyoSwiftTests/__Snapshots__/KlaviyoAPITests/testSuccessfulResponseWithEvent.1.txt:15]: Yeah I tested this using a proxy, it starts at 1. I will update the tests.

**Review by evan-masseau — APPROVED** [review]
LGTM once tests are fixed

**Pull request merged**